File: src/test/scala/com/kjetland/jackson/jsonSchema/testDataScala/PolymorphismOrdering.scala
package com.kjetland.jackson.jsonSchema.testDataScala

import com.fasterxml.jackson.annotation.{JsonSubTypes, JsonTypeInfo}


@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
@JsonSubTypes(Array(
  new JsonSubTypes.Type(value = classOf[PolymorphismOrderingChild3], name = "PolymorphismOrderingChild3"),
  new JsonSubTypes.Type(value = classOf[PolymorphismOrderingChild1], name = "PolymorphismOrderingChild1"),
  new JsonSubTypes.Type(value = classOf[PolymorphismOrderingChild4], name = "PolymorphismOrderingChild4"),
  new JsonSubTypes.Type(value = classOf[PolymorphismOrderingChild2], name = "PolymorphismOrderingChild2")))
trait PolymorphismOrderingParentScala

case class PolymorphismOrderingChild1() extends PolymorphismOrderingParentScala
case class PolymorphismOrderingChild2() extends PolymorphismOrderingParentScala
case class PolymorphismOrderingChild3() extends PolymorphismOrderingParentScala
case class PolymorphismOrderingChild4() extends PolymorphismOrderingParentScala

File: src/test/java/com/kjetland/jackson/jsonSchema/testData/PolymorphismOrdering.java
package com.kjetland.jackson.jsonSchema.testData;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import com.kjetland.jackson.jsonSchema.testData.PolymorphismOrdering.PolymorphismOrderingChild1;
import com.kjetland.jackson.jsonSchema.testData.PolymorphismOrdering.PolymorphismOrderingChild2;
import com.kjetland.jackson.jsonSchema.testData.PolymorphismOrdering.PolymorphismOrderingChild3;
import com.kjetland.jackson.jsonSchema.testData.PolymorphismOrdering.PolymorphismOrderingChild4;


@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
@JsonSubTypes({
  @JsonSubTypes.Type(value = PolymorphismOrderingChild3.class, name = "PolymorphismOrderingChild3"),
  @JsonSubTypes.Type(value = PolymorphismOrderingChild1.class, name = "PolymorphismOrderingChild1"),
  @JsonSubTypes.Type(value = PolymorphismOrderingChild4.class, name = "PolymorphismOrderingChild4"),
  @JsonSubTypes.Type(value = PolymorphismOrderingChild2.class, name = "PolymorphismOrderingChild2")})
public interface PolymorphismOrdering {
    public static class PolymorphismOrderingChild1 implements PolymorphismOrdering {}
    public static class PolymorphismOrderingChild2 implements PolymorphismOrdering {}
    public static class PolymorphismOrderingChild3 implements PolymorphismOrdering {}
    public static class PolymorphismOrderingChild4 implements PolymorphismOrdering {}
}